aidial_adapter_bedrock/llm/model/anthropic.py

```diff
@@ -145,4 +163,127 @@ async def _apredict(
             consumer.append_content(content)
         consumer.close_content()
 
-        consumer.add_usage(compute_usage(prompt, completion))
+        consumer.add_usage(self._compute_usage(prompt, completion))
+
+    def _compute_usage(self, prompt: str, completion: str) -> TokenUsage:
+        batch = self.tokenizer.encode_batch([prompt, completion])
+
+        return TokenUsage(
+            prompt_tokens=len(batch[0].ids),
+            completion_tokens=len(batch[1].ids),
+        )
+
+
+class UsageEventHandler(AsyncMessageStream):
+    def __init__(
+        self,
+        *,
+        cast_to: type[MessageStreamEvent],
+        response: httpx.Response,
+        client: AsyncAnthropic,
+    ):
+        super().__init__(cast_to=cast_to, response=response, client=client)
+        self.prompt_tokens = 0
+        self.completion_tokens = 0
+        self.stop_reason: str | None = None
+
+    async def on_stream_event(self, event: MessageStreamEvent):
+        if isinstance(event, MessageStartEvent):
+            self.prompt_tokens = event.message.usage.input_tokens
+        elif isinstance(event, MessageDeltaEvent):
+            self.completion_tokens += event.usage.output_tokens
+            self.stop_reason = event.delta.stop_reason
+
+
+class AnthropicChat(ChatModel):
+    storage: Optional[FileStorage]
+    client: AsyncAnthropicBedrock
+
+    async def achat(
+        self,
+        consumer: Consumer,
+        params: ModelParameters,
+        messages: List[Message],
+    ):
+        prompt, claude_messages = await to_claude_messages(
+            messages, self.storage
+        )
+        completion_params = {
```

Comment:
Make it `TypedDict` in order to leverage the types from type signatures of `self.client.messages.stream` and `self.client.messages.create`.